packages/yew-functional-macro/src/lib.rs

```diff
@@ -191,21 +199,49 @@ fn function_component_impl(
 
     let ret_type = quote_spanned!(return_type.span()=> ::yew::html::Html);
 
-    let quoted = quote! {
-        #[doc(hidden)]
-        #[allow(non_camel_case_types)]
-        #vis struct #function_name;
+    let quoted = if generics.params.is_empty() {
+        quote! {
+            #[doc(hidden)]
+            #[allow(non_camel_case_types)]
+            #vis struct #function_name;
 
-        impl ::yew_functional::FunctionProvider for #function_name {
-            type TProps = #props_type;
+            impl ::yew_functional::FunctionProvider for #function_name {
+                type TProps = #props_type;
 
-            fn run(#arg) -> #ret_type {
-                #block
+                fn run(#arg) -> #ret_type {
+                    #block
+                }
             }
+
+            #(#attrs)*
+            #vis type #component_name = ::yew_functional::FunctionComponent<#function_name>;
         }
+    } else {
```

Comment:
Is there an issue if we combine both implementations into one?
It seems to me like this should be able to handle empty type parameters.